fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/FDBLuceneQueryTest.java

```diff
@@ -1271,4 +1277,55 @@ void continuations(boolean sorted) throws Exception {
             }
         }
     }
+
+    /**
+     * This tests is reproducing a thread deadlock that happened with Lucene threads when reading schema information.
+     * The fix for the deadlock has removed the {@code Map.computeIfAbsent} call (with the {@code synchronized} block)
+     * that blocked the {@code asyncToSync} calls.
+     * This test creates a large enough set of {@link org.apache.lucene.index.SegmentReader} futures so as to flood the thread pool,
+     * and then queries Lucene. THe large number of futures in the pool, together with the {@code synchronized} block
+     * would cause a deadlock.
+     */
+    @Test
+    void testQueryWithManyDocuments() {
+        String[] words = generateRandomWords(1000);
```

Comment:
I don't understand the api for `generateRandomWords`, but you're not using it correctly here.
IIUC it generates an array of two elements:
`words[1]` is a string of 1,000 randomly generated words, separated by a string
`words[0]` is the last word in `words[1]`

The otherr two usages just use `words[1]` and completely forget `words[0]`.

It might also be simpler by having a `String generateRandomWord(int minLength, int maxLength)`
And then you could have two methods that easily use that:
```
Stream.generate(() -> generateRandomWord(3, 10)).limit(1000).collect(Collectors.joining(" "));
```
and
```
Stream.generate(() -> generateRandomWords(3, 10)).limit(1000).collect(Collectors.toList())
```
and use the later in this test, and then randomly chose from that. Or you could perhaps use the former repeatedly, which would basically result in every document having completely unique sets of words, at least until you generate 141 trillion words. It would be very detached from legitimate use cases of lucene, but might force it to make many segments.

Reply:
Sorry, my mistake. The actual usage of generateRandomWords(500) would generate the string that we eventually need, so no need to process further. The code was the result of having the prior random word selector in place.
Done.